eks/deploy_state.go

```diff
@@ -0,0 +1,92 @@
+package eks
+
+import (
+	"github.com/gruntwork-io/kubergrunt/logging"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
+)
+
+const defaultStateFile = "./.kubergrunt.state"
+
+type DeployState struct {
+	GatherASGInfoDone      bool
+	SetMaxCapacityDone     bool
+	ScaleUpDone            bool
+	WaitForNodesDone       bool
+	CordonNodesDone        bool
+	DrainNodesDone         bool
+	DetachInstancesDone    bool
+	TerminateInstancesDone bool
+	RestoreCapacityDone    bool
+
+	Path string
+	ASG  ASG
```

Comment:
NIT: This state assumes we are updating only a single ASG, which is true as of now, but in the future we want to be able to update multiple ASGs at once (see https://github.com/gruntwork-io/kubergrunt/issues/72). Consider future proofing the state struct to support this.